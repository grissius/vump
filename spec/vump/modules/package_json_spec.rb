require 'vump/modules/package_json'

mod = Vump::PackageJson.new('')
json = File.read(File.expand_path("examples/#{mod.filename}", __dir__))

RSpec.describe Vump::PackageJson do
  it 'filename' do
    expect(mod.filename).to eql('package.json')
  end

  it 'select' do
    expect(mod.select(json)).to eql('0.4.2')
  end

  it 'compose' do
    expect(mod.compose(json, '0.4.3')).to match_snapshot
  end
end
