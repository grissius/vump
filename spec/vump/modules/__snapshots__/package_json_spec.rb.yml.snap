---
"./spec/vump/modules/package_json_spec.rb[1:3]": |
  {
    "name": "your-mom",
    "version": "0.4.3",
    "author": "Jaroslav Šmolík <grissius@gmail.com>",
    "license": "MIT",
    "keywords": [
      "your",
      "mom"
    ]
  }
